### PR TITLE
set up  sphinx doc 

### PR DESCRIPTION
**Pull request by marcobarilari**

### CODE REVIEW

**Comment by marcobarilari** [comment]
Hey @Remi-Gau, I followed the instructions in the spm pipeline repo and something happened but I have no idea about I am doing and if it is working. Sphinx is still very unclear to me

**Comment by Remi-Gau** [comment]
Are you saying you can't figure out the [riddles](https://www.worldbook.com/behind-the-headlines/Mythic-Monday-The-Riddle-of-the-Sphinx) that Sphinx is throwing you?
I guess they have chosen the name correctly. 

Will have a look right away.

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:13]: You are going to need to uncomment those. 

You also need to modify the path in `os.path.abspath` so it points to the root of this repo.

You can just do like in `conf.py` of the CPP_SPM repo

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:34]:
```suggestion
extensions = [
    'sphinxcontrib.matlab', 
    'sphinx.ext.autodoc']
matlab_src_dir = os.path.dirname(os.path.abspath('../../src'))
primary_domain = 'mat'
```

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:33]:
```suggestion

```

File: docs/source/conf.py
Comment (Remi-Gau) [reply docs/source/conf.py:34]: Specify the extensions we will use.

Plus tell the matlab extension where the code is.

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:16]:
```suggestion
import os
import sys
sys.path.insert(0, os.path.abspath('../..'))
```

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:15]:
```suggestion

```

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:14]:
```suggestion

```

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:13]:
```suggestion

```

File: docs/source/conf.py
Comment (Remi-Gau) [inline docs/source/conf.py:50]:
```suggestion
html_theme = 'sphinx_rtd_theme'
```
This defines the general theme and look of the doc.

We will use the ones from read the docs.

**Review by Remi-Gau — CHANGES_REQUESTED** [review]

File: docs/source/index.rst
Comment (Remi-Gau) [inline docs/source/index.rst:13]:
```suggestion
    function_description
```

Adding this here will add an entry to the tree of the table of content and create a link to the `function_descrption.rst` file.

**Review by Remi-Gau — CHANGES_REQUESTED** [review]

**Comment by Remi-Gau** [comment]
Looks like you are on good track to me.

You need to make some changes to the `conf.py`.

Once you are done you can try to build the doc with the following command.

```
sphinx-build -b html source build
```

File: docs/source/conf.py
Comment (marcobarilari) [reply docs/source/conf.py]: I did change these parts but I suspect I made a mistake and ignore the changes while committing. My bad

**Comment by Remi-Gau** [comment]
Also looking around to other repo that use Sphinx it seems that we do not need to commit the content of the `build` folder as read the docs will take care of building the docs. (I am also trying to figure this out).

So you can remove all that of that from the repo and add the `build` folder to `.gitignore`

**Comment by Remi-Gau** [comment]
Oh by the way you can add several comments at once through the github graphic interface. 

I learnt that on the BIDS specs and I still to update their doc there to help others with this. 
https://github.com/bids-standard/bids-specification/pull/633#issuecomment-713388002

**Comment by marcobarilari** [comment]
ok, I have rebuild sphinx doc with the new `config.py` edits and add the `build` folder to .gitignore

File: docs/source/function_description.rst
Comment (marcobarilari) [inline docs/source/function_description.rst:12]: I don't understand this part, what is it the point to have `:func:`initPTB`` and `.. autofunction:: getExperimentStart` after it?

**Comment by marcobarilari** [comment]
I have added all the function in the `function_description.rst` file LMK if looks ok and what's next (apart from making it nicer :) )

File: docs/source/function_description.rst
Comment (Remi-Gau) [reply docs/source/function_description.rst:22]:
```suggestion
```
You are right: this one can be removed and autofunction getExpStart brought up. :-)

**Comment by Remi-Gau** [comment]
> I have added all the function in the `function_description.rst` file LMK if looks ok and what's next (apart from making it nicer :) )

I think you can try using the `read the docs` website to build the docs so we then have an online version.

**Comment by marcobarilari** [comment]
sure, but somehow the sphinx updates should be in a branch (master) of the repo, shouldn't they? Or there is a way to build the rtd from a PR?

**Comment by Remi-Gau** [comment]
If it builds locally you can merge this into `dev`. Then on the read the docs (RTD) in the `advanced settings` you can set the default branch to build the doc from.

See here: https://readthedocs.org/dashboard/cpp-bids-spm/advanced/

Let me know if you can't access I might need to add you as Maintainers on RTD.

You can also start transferring some content from the old markdown `index.md` to the rst files if you want. In this PR before merging and trying to build with RTD or in another PR. Up to you. :rocket:

**Comment by marcobarilari** [comment]
> See here: https://readthedocs.org/dashboard/cpp-bids-spm/advanced/

I see this (lol)

```
        \          SORRY            /
         \                         /
          \    This page does     /
           ]   not exist yet.    [    ,'|
           ]                     [   /  |
           ]___               ___[ ,'   |
           ]  ]\             /[  [ |:   |
           ]  ] \           / [  [ |:   |
           ]  ]  ]         [  [  [ |:   |
           ]  ]  ]__     __[  [  [ |:   |
           ]  ]  ] ]\ _ /[ [  [  [ |:   |
           ]  ]  ] ] (#) [ [  [  [ :===='
           ]  ]  ]_].nHn.[_[  [  [
           ]  ]  ]  HHHHH. [  [  [
           ]  ] /   `HH("N  \ [  [
           ]__]/     HHH  "  \[__[
           ]         NNN         [
           ]         N/"         [
           ]         N H         [
          /          N            \
         /           q,            \
        /                           \
```

**Comment by Remi-Gau** [comment]
oh dear...

Tried to add you using your github username.

Let me know if this works

**Comment by marcobarilari** [comment]
it worked! thank you

**Comment by marcobarilari** [comment]
All right, I added an installation section (also to try sphynx synstax) and going to merge this and publish it to see if it works.

Will add more docs in the future in other PRs, LMK

**Pull request merged**
